### ISSUE TITLE
[Bug]: Inconsistent Application of Dynamic Default Tags to Resources

### ISSUE THREAD

**Issue by jmanno** [hashicorp/terraform-provider-aws#35725]
### Terraform Core Version

1.7.3

### AWS Provider Version

5.35.0

### Affected Resource(s)

All resources with `tags_all` attribute

### Expected Behavior

All resources should have the `tags_all` attribute computed with dynamic default tags applied, reflecting global tagging configurations.

### Actual Behavior

The `tags_all` attribute is not consistently computed across all resources, indicating a failure to apply dynamic default tags as expected.

This is another example of the inconsistent behavior 
![5CACD943-B71B-42C8-A84A-7FE1ABACE2E8](https://github.com/hashicorp/terraform-provider-aws/assets/3864390/c945aa35-d097-448d-a300-f706bfcb68c4)

### Relevant Error/Panic Output Snippet

```shell
❯ terraform plan
aws_vpc.example: Refreshing state... [id=vpc-####]
aws_subnet.example: Refreshing state... [id=subnet-####]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated
with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_subnet.example will be updated in-place
  ~ resource "aws_subnet" "example" {
        id                                             = "subnet-####"
      ~ tags                                           = {
          - "ManagedBy" = "Terraform" -> null
            "Name"      = "my-subnet-resource"
        }
      ~ tags_all                                       = {
          - "ManagedBy" = "Terraform" -> null
            # (1 unchanged element hidden)
        }
        # (15 unchanged attributes hidden)
    }

  # aws_vpc.example will be updated in-place
  ~ resource "aws_vpc" "example" {
        id                                   = "vpc-####"
      ~ tags                                 = {
          - "ManagedBy" = "Terraform" -> null
            "Name"      = "my-vpc-resource"
        }
      ~ tags_all                             = {
          - "ManagedBy" = "Terraform" -> null
            # (1 unchanged element hidden)
        }
        # (14 unchanged attributes hidden)
    }
Plan: 0 to add, 2 to change, 0 to destroy.
```

### Terraform Configuration Files
```terraform
provider "aws" {
  region = "us-east-1"

  default_tags {
    tags = {
        ManagedBy = "Terraform"

        # To reproduce the bug:
        # 1) apply the plan generated by this file
        # 2) uncomment the following line and examine the new plan
        # CreatedAt = formatdate("YYYY-MM-DD", timestamp())
      }
  }
}

terraform {
  required_version = "1.7.3"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "5.35.0"
    }
  }

}

resource "aws_vpc" "example" {
  cidr_block = "10.1.0.0/16"
  tags = {
    Name = "my-vpc-resource"
  }
}

resource "aws_subnet" "example" {
  cidr_block = "10.1.1.0/24"
  vpc_id     = aws_vpc.example.id
  tags = {
    Name = "my-subnet-resource"
  }
}
```

### Steps to Reproduce

   # To reproduce the bug:
        # 1) apply the plan generated by this file
        # 2) uncomment the following line and examine the new plan
        # CreatedAt = formatdate("YYYY-MM-DD", timestamp())

### Important Factoids

We examined a json version of a plan and confirm that terraform is not computing the `tags_all` attribute for some resources. Other resources do have the `tags_all` attribute.

### Would you like to implement a fix?

None

**Comment by jar-b** [comment]
Hello - #30801 documents a similar issue, and [this comment](https://github.com/hashicorp/terraform-provider-aws/issues/30801#issuecomment-1561803934) describes why `default_tags` behave this way with the provided configuration (computed values, such the result of the `timestamp()` function, are not supported due to limitations on how the provider is configured).

Assuming the precise apply time is not a requirement given the formatted value only includes the date portion, the [`plantimestamp` function](https://developer.hashicorp.com/terraform/language/functions/plantimestamp) could provide the desired result. This function is available in Terraform 1.5 and later. As the name suggests, its value reflects the timestamp of the plan operation, which means that during apply the value is known and avoids inclusion of a computed (unknown) value in the `default_tags` block. 

I was able to reproduce the issue with the original configuration, and confirm that changing the following resulted in an apply, followed by a plan with no changes.

```hcl
  default_tags {
    tags = {
      ManagedBy = "Terraform"
      CreatedAt = formatdate("YYYY-MM-DD", plantimestamp())
    }
  }
```
---

```console
% terraform apply -auto-approve

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # aws_subnet.example will be created
  + resource "aws_subnet" "example" {
<snip>
      + tags                                           = {
          + "Name" = "my-subnet-resource"
        }
      + tags_all                                       = {
          + "CreatedAt" = "2024-02-15"
          + "ManagedBy" = "Terraform"
          + "Name"      = "my-subnet-resource"
        }
      + vpc_id                                         = (known after apply)
    }

  # aws_vpc.example will be created
  + resource "aws_vpc" "example" {
<snip>
      + tags                                 = {
          + "Name" = "my-vpc-resource"
        }
      + tags_all                             = {
          + "CreatedAt" = "2024-02-15"
          + "ManagedBy" = "Terraform"
          + "Name"      = "my-vpc-resource"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.
aws_vpc.example: Creating...
aws_vpc.example: Creation complete after 1s [id=vpc-00d35960d803dcb4e]
aws_subnet.example: Creating...
aws_subnet.example: Creation complete after 1s [id=subnet-00e8db1d2863519e1]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

```console
% terraform plan
aws_vpc.example: Refreshing state... [id=vpc-00d35960d803dcb4e]
aws_subnet.example: Refreshing state... [id=subnet-00e8db1d2863519e1]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are
needed.
```

```console
% terraform -v
Terraform v1.7.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v5.35.0
```

Please let me know if I misunderstood any part of the original report given the minimal reproduction may not be reflective of the full configuration included in the screenshots. If this adjustment works for your use case, we can go ahead and close this issue out.

**Comment by jar-b** [comment]
Hey @fcastagninizoom - thanks for your feedback! 

If your configuration and issue varies significantly from this one (computed tags such as a timestamp in the `default_tags` block), it may be worth opening a separate issue with a minimal configuration so we can better assess how to help. If it turns out to be the same root cause as another issue, we can always link and close it as necessary.
